src/main/java/roomescape/infrastructure/JdbcMemberRepository.java

```diff
@@ -0,0 +1,83 @@
+package roomescape.infrastructure;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+import roomescape.auth.Role;
+import roomescape.domain.Member;
+import roomescape.domain.repository.MemberRepository;
+
+@Repository
+public class JdbcMemberRepository implements MemberRepository {
+
+    private static final RowMapper<Member> ROW_MAPPER = (rs, rowNum) ->
+            new Member(
+                    rs.getLong("id"),
+                    rs.getString("name"),
+                    rs.getString("email"),
+                    rs.getString("password"),
+                    Role.valueOf(rs.getString("role"))
```

Comment:
```suggestion
                    Role.from(rs.getString("role"))
```
DB에 저장된 문자열이 Enum에 정의되지 않은 값일 경우 Role.valueOf()는 IllegalArgumentException을 발생시킬 수 있습니다.
DB <-> enum 매핑 실패는 스크립트 오타, 직접 DB 조작, Enum 변경 등 다양한 이유로 발생할 수 있어요.
따라서 팩토리 메서드를 통해 예외를 감싸 유효성 검증을 명시적으로 해주면 좋습니다. 😄

Reply:
넵 반영했습니다!

Reply:
오 맞아요! 핵심을 아주 잘 짚어주셨습니다.
DB <-> enum 간 매핑 실패는 절대 발생해서는 안 되는 오류죠.
이런 경우에는 클라이언트에게는 추상적인 오류 메시지를 전달하고, 개발자가 빠르게 원인을 파악해 수정해야 합니다.
클라이언트의 잘못된 입력으로 발생한 오류라면 구체적인 메시지를 전달해 사용자가 문제를 인지할 수 있도록 하고,
반대로 서버 내부의 문제라면 로그나 예외 클래스에 상세한 정보를 남기되,
클라이언트에게는 “오류가 발생했습니다. 관리자에게 문의해주세요.”와 같은 일반적인 메시지로 처리하는 것이 좋습니다.